File: sample/ios/iosApp/sample/SampleImageResourceProvider.swift
import Foundation
import MetaviewsSample
import Trikot_metaviews

class SampleImageResourceProvider: ImageViewModelResourceProvider {
    func image(fromResource resource: ImageResource?) -> UIImage? {
        if (resource as? ImageResources) == ImageResources.icon {
            return #imageLiteral(resourceName: "icon")
        }
        return nil
    }
}

File: sample/android/src/main/java/com/trikot/viewModels/sample/resource/SampleImageResourceProvider.kt
package com.trikot.viewModels.sample.resource

import android.content.Context
import com.mirego.trikot.viewModels.resource.ImageResource
import com.mirego.trikot.viewModels.resources.ImageViewModelResourceProvider
import com.trikot.viewModels.sample.R

class SampleImageResourceProvider : ImageViewModelResourceProvider {
    override fun resourceIdFromResource(resource: ImageResource, context: Context): Int? {
        if (resource == ImageResources.ICON) {
            return R.drawable.icon
        }
        return null
    }
}
